### PR TITLE
firecloud-ui tests improvements

### PR DESCRIPTION
**Pull request by aweng98**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/billing/BillingSpec.scala [hunk 1]
```diff
@@ -145,7 +145,7 @@ class BillingSpec extends FreeSpec with WebBrowserSpec with UserFixtures with Cl
         if (status == "failure" || status == "unknown") {
           try {
             // click icon to view then capture failure error in screeshot
-            find(xpath(s"//div[@data-test-id='$billingProjectName-row']//span[@data-test-id='status-icon']")).get.underlying.click()
+            find(xpath(s"//div[@data-test-id='$billingProjectName-row']/div[@data-test-class='table-cell']//button")).get.underlying.click()
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/billing/BillingSpec.scala:148]: https://broadworkbench.atlassian.net/browse/WB-221

**Comment by aweng98** [comment]
Jenkins retest

**Comment by aweng98** [comment]
jenkins retest

File: automation/build.sbt [hunk 2]
```diff
@@ -52,9 +52,9 @@ Test / logBuffered := false
   *  setting the limit on Tags.ForkedTestGroup tag, 1 is default.
   *
   *  Warning: can't set too high (parallel execution - tests become flaky)
-  *  This is not number of threads in each JVM. That would be up to sbt.
+  *  This is not number of threads in each JVM. That would be up to sbt to decide.
   */
-Global / concurrentRestrictions := Seq(Tags.limit(Tags.ForkedTestGroup, 3))
+Global / concurrentRestrictions := Seq(Tags.limit(Tags.Test, 25), Tags.limit(Tags.ForkedTestGroup, 3))
```
Comment (aweng98) [inline automation/build.sbt:57]: limit total concurrent test total to <= 25

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Component.scala [hunk 3]
```diff
@@ -56,7 +58,11 @@ abstract class Component(queryString: QueryString)(implicit webDriver: WebDriver
   }
 
   def awaitVisible(): Unit = await visible query
-  def awaitNotVisible(timeout: Long = defaultTimeOutInSeconds): Unit = await notVisible (query, timeout)
+  def awaitNotVisible(timeout: Long = defaultTimeOutInSeconds): Unit = {
+    Try { await notVisible (query, timeout) }.recover {
+      case _: TimeoutException => throw new TimeoutException(s"Failed waiting for notVisible on Element: $query (tried for $timeout seconds).")
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Component.scala:63]: throws meaningful error that describe what happened.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Suggests.scala [hunk 4]
```diff
@@ -53,10 +53,8 @@ trait Suggests extends LazyLogging { this: Component =>
     val listOptionCss1: String = s"""#$dropdownId li *[data-test-id=\"$suggestionTestId\"]"""
     val listOptionCss2: String = s"""[data-test-id=\"$suggestionTestId\"]"""
     val alternateCss = Random.shuffle(List(listOptionCss1, listOptionCss2)).head
-    logger.info(s"(remove) getSuggestionByTestId: alternated CssSelector: $alternateCss")
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Suggests.scala:56]: delete debug msg that used to debug an old issue

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Table.scala [hunk 5]
```diff
@@ -26,7 +26,9 @@ case class Table(queryString: QueryString)(implicit webDriver: WebDriver)
 
   override def awaitReady(): Unit = {
     awaitVisible()
-    await condition { getState == "ready" }
+    Try { await condition { getState == "ready" } }.recover {
+      case _: TimeoutException => throw new TimeoutException(s"Failed waiting for table's data-test-state == ready ($query)")
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Table.scala:30]: meaningful error msg

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/DownloadUtil.scala [hunk 6]
```diff
@@ -17,24 +17,13 @@ import org.scalatest.time.{Seconds, Span}
 
 trait DownloadUtil extends Eventually with LazyLogging with WebBrowserUtil {
 
-
   def makeTempDownloadDirectory(): String = {
-    /*
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/DownloadUtil.scala:22]: delete old comments that described a resolved issue

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataUseAwareSearchSpec.scala [hunk 7]
```diff
@@ -54,7 +54,10 @@ class DataUseAwareSearchSpec extends FreeSpec with WebBrowserSpec with UserFixtu
         val ffiSuggestionId = "suggestion-http://purl.obolibrary.org/obo/DOID_0050433"  // fatal familial insomnia
         val ffiTagId = "doid:0050433-tag"
 
-        researchPurposeModal.enterOntologySearchText("fatal").exists(_.contains(ffiSuggestionText)) shouldBe true
+        eventually {
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataUseAwareSearchSpec.scala:57]: wrap call in `eventually`

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/workspace/WorkspaceWriterSpec.scala [hunk 8]
```diff
@@ -29,7 +29,7 @@ class WorkspaceWriterSpec extends FreeSpec with Matchers with WebBrowserSpec wit
       "should only see the estimated monthly storage fee in the Project Cost section of the summary page" in {
         val user = UserPool.chooseStudent
         implicit val authToken: AuthToken = authTokenOwner
-        val testName = "WorkspaceWriterSpec_writerAccess_projectCost"
+        val testName = "WorkspaceWriterAccess"
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/workspace/WorkspaceWriterSpec.scala:32]: use shorter string for workspace name. long string doesn't add any benefit and harder to read in test log.

**Review by gdlex4015 — APPROVED** [review]

**Comment by aweng98** [comment]
jenkins retest

**Pull request merged**